controller/node_controller.go

```diff
@@ -354,3 +361,35 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 
 	return nil
 }
+
+func (nc *NodeController) syncNodeStatus(pod *v1.Pod, node *longhorn.Node) error {
+	// sync bidirectional mount propagation for node status to check whether the node could deploy CSI driver
+	for _, mount := range pod.Spec.Containers[0].VolumeMounts {
+		if mount.Name == types.LonghornSystemKey {
+			mountPropagationStr := ""
+			if mount.MountPropagation == nil {
+				mountPropagationStr = "nil"
+			} else {
+				mountPropagationStr = string(*mount.MountPropagation)
+			}
+			if mount.MountPropagation == nil || *mount.MountPropagation != v1.MountPropagationBidirectional {
+				logrus.Debugf("The MountPropagation value %s is not expected from pod %s, node %s", mountPropagationStr, pod.ObjectMeta.Name, pod.Spec.NodeName)
```

Comment:
This log will continuing print if mount propagation wasn't enabled? Sounds like a bad idea to flood the log. Let's print the log only if the newly detected status is different from the value stored in the `node.Status.MountPropagation` (which covers both value is true and false).